lms/djangoapps/course_api/blocks/transformers/student_view.py

```diff
@@ -14,6 +14,11 @@ class StudentViewTransformer(BlockStructureTransformer):
     READ_VERSION = 1
     STUDENT_VIEW_DATA = 'student_view_data'
     STUDENT_VIEW_MULTI_DEVICE = 'student_view_multi_device'
+    STUDENT_VIEW_DATA_CONTEXT = {
+        'video': {
```

Comment:
Can this be implemented without infiltrating the generic core API with block-specific details?
Why not have the video-block know which contexts it should be sending back? That is, just include `STUDENT_VIEW_DATA_CONTEXT[video]` within the video block's implementation.

Reply:
@nasthagiri You mean making [VideoBlock](https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/video_module/video_module.py#L1084) send all encodings by default and then transforming it in [VideoBlockURLTransformer](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/course_api/blocks/transformers/video_urls.py#L15)?

Reply:
Yes, that works.
Also, VideoBlockURLTransformer probably should live outside of the `course_api` Django app - in Video's own Django app/folder, along with `video_module.py`.

Reply:
@nasthagiri I have moved STUDENT_VIEW_DATA_CONTEXT[video] inside video_module.py. VideoBlockURLTransformer contains blocks api related attributes like student_view_data and StudentViewTransformer. Also it isn't used by any package outside of course_api.